src/test/java/com/ot/restaurant/repositories/CustomerRepositoryImplTest.java

```diff
@@ -0,0 +1,83 @@
+package com.ot.restaurant.repositories;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.ot.restaurant.builders.CustomerBuilder;
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.IdNullException;
+import com.ot.restaurant.fixtures.CustomerFixture;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerRepositoryImplTest {
+  private CustomerRepositoryImpl customerRepository = new CustomerRepositoryImpl();
+  private List<Customer> customerListTest = new ArrayList<>();
+  private List<Status> statuses = new ArrayList<>();
+
+  @BeforeEach
+  void setUp() throws Exception {
+
+    final var customers =
+        List.of(
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki1").build()),
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki2").build()),
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki3").build()));
+    customers.forEach(customerRepository::insert);
+    customerListTest.addAll(customers);
+    statuses.add(Status.ACTIVE);
+  }
+
+  @Test
+  void shouldReturnTheCustomerListWhenFindAll() {
+    assertEquals(customerListTest, customerRepository.findAll(statuses));
+  }
+
+  @Test
+  void shouldReturnSpecificCustomerWhenFindByIdAndStatusActive() throws Exception {
+    final var existingCustomer = customerRepository.findById(1L, Status.ACTIVE).get();
+
+    assertEquals(1L, existingCustomer.getId());
+    assertEquals("Loki2", existingCustomer.getFirstName());
+  }
+
+  @Test
+  void shouldTrowsExceptionWhenFindByIdWithNullId() throws Exception {
+    assertThrows(
+        IdNullException.class,
+        () -> {
+          customerRepository.findById(null, Status.ACTIVE);
+        });
+  }
+
+  @Test
+  void shouldAddCustomerWhenInsert() throws Exception {
+    final var customer =
+        CustomerFixture.buildCustomerFromExample(
+            new CustomerBuilder().withFirstName("Samahia").build());
+    customerRepository.insert(customer);
+    final var existingCustomers = customerRepository.findAll(statuses);
+
+    assertEquals(
+        existingCustomers.getLast(),
+        customerRepository
+            .findById(existingCustomers.getLast().getId(), existingCustomers.getLast().getStatus())
+            .get());
+  }
+
+  @Test
+  void shouldSetStatusToDeletedWhenDeleteById() throws Exception {
+    final var existingCustomers = customerRepository.findAll(statuses);
+    final var customerAddedId =  existingCustomers.getLast();
+
+    customerRepository.deleteById(customerAddedId.getId(), customerAddedId);
+    statuses.add(Status.DELETED);
```

Comment:
Avoid modifying shared variables because all tests can run in parallel and one test can be affected by others. If you want to use the statuses why not to simple declare it on each test, for example:

```
customerRepository.findAll(List.of(Status.DELETED))
```

Also why don't you use function overload?

You could go to your repo and do:
```
For interface:

List<Customer> findAll() throws Exception;
List<Customer> findAll(List<Status> statuses) throws Exception;

For impl:

List<Customer> findAll() throws Exception {
  return findAll(List.of(Status.ACTIVE))
}

List<Customer> findAll(List<Status> statuses) throws Exception {
   ......
}
```

So you will be able to decide if you want to use either `findAll() or findAll(statuses)`